Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBufferSetProxy.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if ENABLE(GPU_PROCESS)
+
+namespace WebKit {
+
+// A RemoteRenderingBufferSet is a set of three ImageBuffers (front, back,
+// secondary back) owned by the GPU process, for the purpose of drawing
+// successive (layer) frames.
+// To draw a frame, the consumer allocates a new RemoteImageBufferProxy and
+// asks the RemoteRenderingBuffer set to map it to an appropriate new front
+// buffer (either by picking one of the back buffers, or by allocating a new
+// one). It then copies across the pixels from the previous front buffer,
+// clips to the dirty region and clears that region.
+// Usage is done through RemoteRenderingBackendProxy::prepareBuffersForDisplay,
+// so that a Vector of RemoteRenderingBufferSets can be used with a single
+// IPC call.
+class RemoteRenderingBufferSetProxy : public RefCounted<RemoteRenderingBufferSetProxy>, public CanMakeWeakPtr<RemoteRenderingBufferSetProxy> {
```

Comment:
I think this class should be RemoteImageBufferSetProxy.
Could this just reference the actual ImageBuffers (or RemoteImageBufferProxy instances) that it references?